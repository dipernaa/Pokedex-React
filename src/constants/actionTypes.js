import { buildActionNames } from '../actions/utils/asyncUtils';

// pokemon
export const GET_POKEMON = buildActionNames('GET_POKEMON');
export const GET_POKEMON_DETAILS = buildActionNames('GET_POKEMON_DETAILS');
