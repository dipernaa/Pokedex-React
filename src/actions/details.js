import api from './utils/api';
import { createAsyncAction } from './utils/asyncUtils';
import { GET_POKEMON_DETAILS } from '../constants/actionTypes';

function getPokemonDetails(url) {
  return createAsyncAction(GET_POKEMON_DETAILS, () => api.getAsync(`/api/details?url=${url}`));
}

export default {
  getPokemonDetails
};
