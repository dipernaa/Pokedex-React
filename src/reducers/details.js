import { GET_POKEMON_DETAILS } from '../constants/actionTypes';

export const initialState = {
  results: null,
  isLoading: false
};

export default function(state = initialState, action) {
  console.log(state);
  console.log(action);
  switch (action.type) {
    case GET_POKEMON_DETAILS.request:
      return {
        ...state,
        isLoading: true
      };

    case GET_POKEMON_DETAILS.failure:
      return {
        ...state,
        isLoading: false
      };

    case GET_POKEMON_DETAILS.success:
      return {
        ...state,
        results: action.data,
        isLoading: false
      };

    default:
      return state;
  }
}
