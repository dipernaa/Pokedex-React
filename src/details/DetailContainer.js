import React, { Component } from 'react';
import { connect } from 'react-redux';
import muiThemeable from 'material-ui/styles/muiThemeable';
import Loading from '../shared/Loading';
import PokemonDetail from './PokemonDetail';
import { getPokemonDetails } from '../actions';

export class DetailContainer extends Component {

  componentWillMount() {
    const { url } = this.props.location.query;
    this.props.getPokemonDetail(url);
  }

  render() {
    const { isLoading, detailData } = this.props;

    if (isLoading) {
      return <Loading />;
    }

    return (
      <PokemonDetail pokemon={detailData}/>
    );
  }
}

export const mapStateToProps = (state) => {
  const { details } = state;

  return {
    isLoading: details.isLoading,
    detailData: details.results
  };
};

export const mapDispatchToProps = (dispatch) => ({
  getPokemonDetail: (url) => dispatch(getPokemonDetails(url))
});

export default muiThemeable()(connect(mapStateToProps, mapDispatchToProps)(DetailContainer));
