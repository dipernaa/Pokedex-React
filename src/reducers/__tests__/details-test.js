import details, { initialState } from '../details';
import { GET_POKEMON_DETAILS } from '../../constants/actionTypes';

it('handles unexpected action', () => {
  expect(details(initialState, { type: '_NULL' }))
    .toMatchSnapshot();
});

it('handles GET_POKEMON_DETAILS_REQUEST action', () => {
  expect(details(initialState, { type: GET_POKEMON_DETAILS.request }))
    .toMatchSnapshot();
});

it('handles GET_POKEMON_DETAILS_SUCCESS action', () => {
  const pokemonDetails = {
    name: 'pokedude',
    id: 34
  };

  expect(details(initialState, { type: GET_POKEMON_DETAILS.success, data: pokemonDetails }))
    .toMatchSnapshot();
});

it('handles GET_POKEMON_DETAILS_FAILURE action', () => {
  expect(details(initialState, { type: GET_POKEMON_DETAILS.failure }))
    .toMatchSnapshot();
});
