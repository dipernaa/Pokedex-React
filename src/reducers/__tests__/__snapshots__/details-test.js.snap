// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles GET_POKEMON_DETAILS_FAILURE action 1`] = `
Object {
  "isLoading": false,
  "results": null,
}
`;

exports[`handles GET_POKEMON_DETAILS_REQUEST action 1`] = `
Object {
  "isLoading": true,
  "results": null,
}
`;

exports[`handles GET_POKEMON_DETAILS_SUCCESS action 1`] = `
Object {
  "isLoading": false,
  "results": Object {
    "id": 34,
    "name": "pokedude",
  },
}
`;

exports[`handles unexpected action 1`] = `
Object {
  "isLoading": false,
  "results": null,
}
`;
